pages/thc/[processRef]/manager-review.tsx

```diff
@@ -0,0 +1,314 @@
+import getProcessRef from "../../../helpers/getProcessRef";
+import { ProcessStage } from "../../../helpers/ProcessStage";
+import useDatabase from "../../../helpers/useDatabase";
+import useDataSet from "../../../helpers/useDataSet";
+import useDataValue from "../../../helpers/useDataValue";
+import MainLayout from "../../../layouts/MainLayout";
+import PageTitles from "../../../steps/PageTitles";
+import Storage from "../../../storage/Storage";
+
+const ReviewPage: NextPage = () => {
+  const router = useRouter();
+  const processRef = getProcessRef(router);
+  const processDatabase = useDatabase(Storage.ProcessContext);
+
+  const tenants = useDataValue(
+    Storage.ExternalContext,
+    "residents",
+    processRef,
+    (values) => (processRef ? values[processRef]?.tenants : undefined)
+  );
+
+  const tenantsValue = tenants.result || [];
+
+  const tenantIdsPresentForCheck = useDataValue(
+    Storage.ProcessContext,
+    "tenantsPresent",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const tenantIdsPresentForCheckValue = tenantIdsPresentForCheck.result || [];
+
+  const tenantsWithPresentStatus = tenantsValue.map((tenant) => ({
+    ...tenant,
+    present: tenantIdsPresentForCheckValue.includes(tenant.id),
+  }));
+
+  const address = useDataValue(
+    Storage.ExternalContext,
+    "residents",
+    processRef,
+    (values) => (processRef ? values[processRef]?.address : undefined)
+  );
+
+  const tenureType = useDataValue(
+    Storage.ExternalContext,
+    "tenancy",
+    processRef,
+    (values) => (processRef ? values[processRef]?.tenureType : undefined)
+  );
+
+  const tenancyStartDate = useDataValue(
+    Storage.ExternalContext,
+    "tenancy",
+    processRef,
+    (values) => (processRef ? values[processRef]?.startDate : undefined)
+  );
+
+  const officerFullName = useDataValue(
+    Storage.ExternalContext,
+    "officer",
+    processRef,
+    (values) => (processRef ? values[processRef]?.fullName : undefined)
+  );
+
+  const officerFullNameValue = officerFullName.result || "Loading...";
+
+  const submittedDate = useDataValue(
+    Storage.ProcessContext,
+    "submitted",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const submittedDateValue = submittedDate.result
+    ? formatDate(new Date(submittedDate.result), "d MMMM yyyy")
+    : "Loading...";
+
+  const isUnannouncedVisit = useDataValue(
+    Storage.ProcessContext,
+    "isUnannouncedVisit",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const getManagerSummaryText = (
+    yesValue: string,
+    noValue: string,
+    databaseValue: { value: string; notes: Notes }
+  ): string => {
+    if (databaseValue.value === "yes") {
+      return yesValue;
+    } else {
+      return databaseValue.notes.length > 0
+        ? `${noValue}. Note: ${databaseValue.notes[0].value}`
+        : `${noValue}`;
+    }
+  };
```

Comment:
Let's pull this outside the component scope as it doesn't rely on anything by closure. Makes the render easier to follow.